### ISSUE TITLE
"Today" data broken and no cards on New Year's Eve.

### ISSUE THREAD

**Issue by thatdude624** [ankidroid/Anki-Android#4028]
The "Today" section in stats always shows 0 for all fields for me, as does the text at the bottom of the main menu: "Studied 0 cards in 0 minutes" even after having completed the day's cards.

Sync with Anki Desktop is quite broken, and I almost always need to force changes in one direction.
I've had these strange issues for a while (about half a year when I first got Anki, so it's not related to the new UI). I almost exclusively use Ankidroid, only using Sync and desktop Anki for major changes to my decks.

This wasn't too annoying until today, the last day of the year, where I apparently have no cards due today even though I should. My guess is that some parts of the program think the date is off by 1, making today the 366th day of a non-leap year, which has no cards due since it doesn't exist.

Timezone is GMT +2, clock on both Ankidroid and Anki Desktop is correct.

AnkiDroid Version = 2.5.4

Android Version = 4.4.2

ACRA UUID = cd2acd01-f827-42e5-bf79-76ea79117e3e

**Comment by eginhard** [comment]
Can you try reinstalling AnkiDroid?

**Comment by thatdude624** [comment]
Well, oddly enough, the problem completely fixed itself on Jan 1st. I did flash cards from Jan 1st on New Years eve, leaving Jan 1st only with the cards that I got wrong the previous day, but after every subsequent day Anki's working better than ever.

Sync works flawlessly and no longer complains about inconsistencies. The "cards done today" counters actually work. The "forecast" graph shifted one to the left while the cards done and time spent on card graphs shifted by one day as well. Anki actually automatically tells me what cards are due today on opening the app instead of me first having to open a deck before the numbers update. All the weird behavior just fixed itself.

Well, sorry for bothering you, but hopefully this might be useful for when someone else experiences a similar problem. I'm just glad it's all working fine again!

**Comment by timrae** [comment]
One thing to note about the due today count, is that it will show 0 if you
haven't done any reviews in the last 7 days (or if the reviews haven't been
synced in yet)

**Comment by thatdude624** [comment]
That's an interesting feature, but I do my reviews almost every day, missing at most one day (and even then I would still attempt at least a few cards).
Before the UI change, I experienced a similar related bug that sometimes, cards that I got wrong would sometimes be "forgotten" about (they wouldn't show up again after a few seconds/minutes like they usually do, even though the red counter was accurate) until I either finished all due cards in green or closed the deck with the back button and opened it again, which would then let me do those cards. This only happened sometimes, and this bug was fixed with the new UI change (replaced with only the minor due today update problem, which fixed itself for me on Jan 1st), but it seems to be related.
